### PR TITLE
[C++][Ue4] various bus fixes

### PR DESCRIPTION
**Pull request by Kahncode**
More fixes to UE4 code generator. Tested compilation under various project setups, including linux compilation. Added support for string and enum type definitions. Better handling of variable names in edge cases, should no longer produce empty variable names.

### PR checklist

- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] If contributing template-only or documentation-only changes which will change sample output, [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) before.
- [x] Run the shell script(s) under `./bin/` (or Windows batch scripts under`.\bin\windows`) to update Petstore samples related to your fix. This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit, and these must match the expectations made by your contribution. You only need to run `./bin/{LANG}-petstore.sh`, `./bin/openapi3/{LANG}-petstore.sh` if updating the code or mustache templates for a language (`{LANG}`) (e.g. php, ruby, python, etc).
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master`, `4.3.x`, `5.0.x`. Default: `master`.
- [x] Copy the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) to review the pull request if your PR is targeting a particular programming language.

### CODE REVIEW

File: modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/CppUE4ClientCodegen.java [hunk 1]
```diff
@@ -497,13 +497,20 @@ public String toVarName(String name) {
...
+
+        //Avoid empty variable name at all costs
+        if(!camelCaseName.isEmpty()) {
+            name = camelCaseName;
+        }
+
         // for reserved word or word starting with number, append _
         if (isReservedWord(name) || name.matches("^\\d.*")) {
             name = escapeReservedWord(name);
         }
```
Comment (wing328) [inline modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/CppUE4ClientCodegen.java:511]: I think the special handling of the reserved words may not work here (before and after this PR).

I'll file another PR to address it later.

**Comment by wing328** [comment]
>  Better handling of variable names in edge cases

Do you have an example of the edge cases? We may add it to the test spec.

**Comment by Kahncode** [comment]
Hi @wing328 The edge case encountered was "_". Because of camelize, this generates an empty variable name. Also in the previous logic, it used underscores to escape unique names and language reserved keywords, but before camelizing, so it would end up removing the underscores in the camelize step, and create invalid variable names.

**Comment by wing328** [comment]
OK. Likely generators have similar issues handling just "_". Will add it to the test spec later and update other generators accordingly
